modules/lvgl/input/lvgl_pointer_input.c

```diff
@@ -46,62 +54,56 @@ static void lvgl_pointer_process_event(const struct device *dev, struct input_ev
 		return;
 	}
 
-	/* adjust coordinates */
-	if (cfg->swap_xy) {
-		lv_coord_t tmp;
-
-		tmp = point->x;
-		point->x = point->y;
-		point->y = tmp;
-	}
+	/* prevent re-inversion of the coordinates on touch release */
+	if (data->pending_event.state == LV_INDEV_STATE_PR) {
```

Comment:
@fabiobaltieri Regarding the comment you left in the issue: You are right, maybe it would be easier to understand if I handle everything within the switch.

Reply:
sorry did not realize that the PR was already out, yeah up to you but I think it'd be cleaner so you don't depend on the event state

Reply:
No prob, feedback is always appreciated :+1: . Drafted the solution in a subsequent commit. I am unsure if the rotation should be outside of the switch. If I follow it right, the current draft would produce weird results on a flipped screen, right?